third_party/xla_client/mhlo_helper.cc

```diff
@@ -1,18 +1,33 @@
 #include "third_party/xla_client/mhlo_helper.h"
+#include "third_party/xla_client/xla_util.h"
 #include "tensorflow/compiler/xla/translate/hlo_to_mhlo/hlo_to_mlir_hlo.h"
 #include "tensorflow/compiler/xla/translate/mhlo_to_hlo/mlir_hlo_to_hlo.h"
+#include "mlir/IR/Verifier.h"  // from @llvm-project
+
+#include <iostream>
 
 namespace xla {
 
 void hlo_mhlo_hlo_roundtrip_helper(HloModuleProto* proto) {
   mlir::MLIRContext context;
   mlir::ModuleOp mlir_module = mlir::ModuleOp::create(mlir::UnknownLoc::get(&context));
   auto status = ConvertHloToMlirHlo(mlir_module, proto, /*import_all_computations=*/false);
+  if (!status.ok()) {
+    std::cout << "hlo2mhlo not ok" << std::endl;
+  }
+  if (!mlir::verify(mlir_module).succeeded()) {
+    std::cout << "verify not ok" << std::endl;
+    TF_ASSIGN_OR_RETURN(auto hlo_module, xla::util::CreateModuleFromProto(*proto));
+    std::cout << hlo_module->ToString() << std::endl;
```

Comment:
(I think) you should also be able to dump the invalid MLIR using `mlir_module.dump()`. Would probably be helpful as well.

Also should we return in this `if`? Log invalid case and return the original HLO before swapping. That would allow us to get all the failing test HLOs if a crash or anything is blocking them from running.

Reply:
Oh nice! Both failing on dynamic reshape?

Reply:
Yes, I think they are the same case with minor difference. The error messages are the same